Source/WebCore/dom/Element.cpp

```diff
@@ -3809,23 +3763,13 @@ ExceptionOr<RefPtr<Attr>> Element::setAttributeNode(Attr& attrNode)
 
 ExceptionOr<RefPtr<Attr>> Element::setAttributeNodeNS(Attr& attrNode)
 {
-    RefPtr oldAttrNode = attrIfExists(attrNode.qualifiedName());
-    if (oldAttrNode == &attrNode)
-        return oldAttrNode;
-
-    // InUseAttributeError: Raised if node is an Attr that is already an attribute of another Element object.
-    // The DOM user must explicitly clone Attr nodes to re-use them in other elements.
-    if (attrNode.ownerElement() && attrNode.ownerElement() != this)
-        return Exception { ExceptionCode::InUseAttributeError };
-
     // Attr::value() will return its 'm_standaloneValue' member any time its Element is set to nullptr. We need to cache this value
     // before making changes to attrNode's Element connections.
     auto attrNodeValue = attrNode.value();
-    unsigned index = 0;
 
     if (document().requiresTrustedTypes()) {
         auto attributeTypeAndSink = trustedTypeForAttribute(nodeName(), attrNode.qualifiedName().localName(), this->namespaceURI(), attrNode.qualifiedName().namespaceURI());
-        auto attributeNodeValue = trustedTypesCompliantAttributeValue(attributeTypeAndSink.attributeType, attrNodeValue, this, attributeTypeAndSink.sink);
+        auto attributeNodeValue = trustedTypesCompliantAttributeValue(*document().scriptExecutionContext(), attributeTypeAndSink.attributeType, attrNodeValue, attributeTypeAndSink.sink);
```

Comment:
I think it will help here too.
```suggestion
        auto compliantValue = trustedTypesCompliantAttributeValue(document(), type.attributeType, attrNodeValue, type.sink);
```